lib.go

```diff
@@ -138,6 +140,7 @@ func (vm *VM) Instantiate(
 		return nil, gasUsed, err
 	}
 
+	gasUsed += uint64(deserCost.Mul(len(data)).Int())
```

Comment:
Hmm, I now realize I do not completely understand how the `GasMeter` is used. It seems to be passed to the database only.

`gasUsed` is measured in Wasmer gas points, which are different than Cosmos SDK gas points. If we use the implementation here as shown, `deserCost` is also in Wasmer gas points. I think this is okay.

Now this code does not stop execution if the user does not have sufficient gas left. So an additional check like this is needed

```suggestion
	gasForDeserialization := uint64(deserCost.Mul(len(data)).Int())
	if gasLimit < gasForDeserialization + gasUsed {
		// return gas depletion error
	}
	gasUsed += gasForDeserialization
```